src/app/clusters/fan-control-server/fan-control-server.cpp

```diff
@@ -346,8 +346,9 @@ void MatterFanControlClusterServerAttributeChangedCallback(const app::ConcreteAt
             VerifyOrReturn(EMBER_ZCL_STATUS_SUCCESS == status,
                            ChipLogError(Zcl, "Failed to get SpeedSetting with error: 0x%02x", status));
 
-            float percent        = percentSetting.Value();
-            uint8_t speedSetting = static_cast<uint8_t>(ceil(speedMax * (percent * 0.01)));
+            float percent = percentSetting.Value();
+            // Minus insignificant number 0.00000001 before ceil() to avoid floating point precision error
+            uint8_t speedSetting = static_cast<uint8_t>(ceil(speedMax * (percent * 0.01) - 0.00000001));
```

Comment:
If we use `ceilf` instead of `ceil`, that seems to fix the specific problem with 10 and 70, but could it fail elsewhere?  That will avoid the "15th decimal" problem, since float can't represent that many digits of precision, but without us making up an ad-hoc delta to support from the value....

Reply:
Actually, https://github.com/project-chip/connectedhomeip/pull/27912#discussion_r1261969105 is probably the way to do this.